patches/api/0457-Paper-Item-API.patch

```diff
@@ -0,0 +1,582 @@
++    PropertyKey<Integer> REPAIR_COST = PropertyKey.of("RepairCost", Integer.class);
++    PropertyKey<Integer> CUSTOM_MODEL_DATA = PropertyKey.of("CustomModelData", Integer.class);
++    PropertyKey<Integer> DAMAGE = PropertyKey.of("Damage", Integer.class);
++    PropertyKey<List<AttributePair>> ATTRIBUTES = PropertyKey.of("AttributeModifiers", TypeFactory.parameterizedClass(List.class, AttributePair.class));
++    PropertyKey<MutablePropertyView> BLOCK_STATE = PropertyKey.of("BlockStateTag", MutablePropertyView.class);
++    PropertyKey<MutablePropertyView> ENTITY_TAG = PropertyKey.of("EntityTag", MutablePropertyView.class);
++
++    // DISPLAY PROPERTIES
++    PropertyKey<MutablePropertyView> DISPLAY = PropertyKey.of("display", MutablePropertyView.class);
++
++    PropertyKey<Component> DISPLAY_NAME = PropertyKey.of("Name", Component.class);
++    PropertyKey<List<? extends Component>> LORE = PropertyKey.of("Lore", TypeFactory.parameterizedClass(List.class, Component.class));
++    PropertyKey<Color> COLOR = PropertyKey.of("color", Color.class);
++    PropertyKey<Color> MAP_COLOR = PropertyKey.of("MapColor", Color.class);
++
++    // Armor Meta
++    PropertyKey<ArmorTrim> ARMOR_TRIM = PropertyKey.of("Trim", ArmorTrim.class);
++
++    // TropicalFishBucketMeta
++    PropertyKey<TropicalFishVariant> TROPICAL_FISH_BUCKET_VARIANT = PropertyKey.of("BucketVariantTag", TropicalFishVariant.class);
++
++    // SuspiciousStewMeta
++    PropertyKey<List<StewEffect>> STEW_EFFECTS = PropertyKey.of("effects", TypeFactory.parameterizedClass(List.class, StewEffect.class));
++
++    // CraftMetaSkull
++    // Can be a STRING or a skull profile
++    PropertyKey<PlayerProfile> SKULL_OWNER = PropertyKey.of("SkullOwner", PlayerProfile.class);
++    PropertyKey<String> SKULL_OWNER_NAME = PropertyKey.of("SkullOwner", String.class);
++
++    // PotionMeta
++    PropertyKey<Color> CUSTOM_POTION_COLOR = PropertyKey.of("CustomPotionColor", Color.class);
++    PropertyKey<PotionType> DEFAULT_POTION_TYPE = PropertyKey.of("Potion", PotionType.class);
++    PropertyKey<List<PotionEffect>> CUSTOM_POTION_EFFECTS = PropertyKey.of("CustomPotionEffects", TypeFactory.parameterizedClass(List.class, PotionEffect.class));
++
++    // MapMeta
++    PropertyKey<Integer> MAP_ID = PropertyKey.of("map", Integer.class);
++    PropertyKey<Byte> MAP_SCALE_DIRECTION = PropertyKey.of("map_scale_direction", Byte.class);
++    // Unused, really. Marks the map to be locked when crafted
++    PropertyKey<Boolean> MAP_TO_LOCK = PropertyKey.of("map_to_lock", Boolean.class);
++
++    // KnowledgeBookMeta
++    PropertyKey<List<NamespacedKey>> RECIPES = PropertyKey.of("Recipes", TypeFactory.parameterizedClass(List.class, NamespacedKey.class));
++
++    // FireworkMeta
++    PropertyKey<MutablePropertyView> FIREWORKS = PropertyKey.of("Fireworks", MutablePropertyView.class);
```

Comment:
Same thing applies here, only have properties which point directly to specific attributes of fireworks